File: findLonely_num.cpp
#include <iostream>
using namespace std;
int main()
{
    int  i,n;
    cout<<"enter limit for array :"<<endl;
    cin>>n;

    int a[n];
    cout<<"enter array elements :"<<endl;
    for(int i=0;i<n;i++){
            cin>>a[i];
    }
    cout<<"lonly numbers are :"<<endl;
    for (int i = 1; i < n-1; i++)
    {
        if (a[i] == a[i - 1] + 1 || a[i] == a[i - 1] - 1 || a[i] == a[i + 1] + 1 || a[i] == a[i + 1] - 1)
        {
            continue;
        }
        else
        {
            cout << a[i]<<"\t";
        }
    }
}

File: findLonely_num.js

let a=[10, 6, 9, 8, 3, 7, 6],b=[];
console.log("lonly number :\n");
for (let i = 1; i < 6; i++)
{
    if (a[i] == a[i - 1] + 1 || a[i] == a[i - 1] - 1 || a[i] == a[i + 1] + 1 || a[i] == a[i + 1] - 1)
    {
        continue;
    }
    else
    {
       b.push(a[i]);    
    }
}

console.log(b);